policy/approval/approve.go

```diff
@@ -256,7 +257,7 @@ func (r *Rule) filteredCandidates(ctx context.Context, prctx pull.Context) ([]*c
 	}
 	sort.Stable(common.CandidatesByCreationTime(candidates))
 
-	if !r.Options.InvalidateOnPush {
+	if !r.Options.IgnoreEditedComments && !r.Options.InvalidateOnPush {
```

Comment:
I think we need to treat these as two separate conditions: the code on L264-275 is only relevant for `InvalidateOnPush`, so ideally we only run that when the option is enabled. There's also a situation (L269) where we could return early without applying `IgnoreEditedComments` filtering.

It might make sense to extract each filter into a new helper function so that we can directly update `candidates`, something like:

```go
candidates, err := r.Options.GetMethods().Candidates(ctx, prctx)
// ...

if r.Options.IgnoreEditedComments {
  candidates, err = r.filterEditedCandidates(ctx, prctx, candidates)
  if err != nil {
    return nil, err
  }
}
if r.Options.InvalidateOnPush {
  candidates, err = r.filterInvalidCandidates(ctx, prctx, candidates)
  if err != nil {
    return nil, err
  }
}

return candidates
```

Reply:
i see, i missed L269, will give this approach a shot.

Reply:
i guess i was just trying to avoid looping over the candidates multiple times